src/main/java/factionsplusplus/models/Group.java

```diff
@@ -47,46 +53,105 @@ public void setDescription(String newDesc) {
+        if (! this.isMember(playerUuid)) this.addMember(playerUuid);
+
+        GroupMember member = this.getMember(playerUuid);
+        if (! member.hasRole(GroupRole.Owner)) {
+            member.addRole(GroupRole.Owner);
+        }
+    }
+
+    public boolean hasOwner() {
+        return this.getOwner() != null;
+    }
+
+    /**
+     * Add a new member to the Map with an associated GroupMember object.
+     * <p>
+     * The playerUUID provided was be stored in a HashMap to allow for fast
+     * lookup. This will gurantee that a GroupMember object is assocated with
+     * every member in the group.
+     *
+     * @param playerUuid
+     */
+    public void addMember(UUID playerUuid) {
+        GroupMember member = new GroupMember(playerUuid);
+        member.addRole(GroupRole.Member);
+        members.add(new GroupMember(playerUuid));
+
     }
 
-    public void addMember(UUID UUID) {
-        members.add(UUID);
+    /**
+     * Add a new member to the Map with an associated GroupMember object.
+     * <p>
+     * The playerUUID provided was be stored in a HashMap to allow for fast
+     * lookup. This will gurantee that a GroupMember object is assocated with
+     * every member in the group. The provided role will be stored in an
+     * ArrayList of roles.
+     *
+     * @param playerUuid
+     * @param role
+     */
+    public void addMember(UUID playerUuid, GroupRole role) {
+        GroupMember member = new GroupMember(playerUuid);
+        member.addRole(role);
+        members.add(member);
     }
 
-    public void removeMember(UUID UUID) {
-        members.remove(UUID);
+    public GroupMember getMember(UUID playerUuid) {
+        return this.members
+            .stream()
+            .filter(member -> member.getId() == playerUuid)
+            .findFirst()
+            .orElse(null);
     }
 
-    public boolean isMember(UUID uuid) {
-        return members.contains(uuid);
+    public void removeMember(UUID playerUuid) {
+        members.remove(this.getMember(playerUuid));
     }
 
-    public List<UUID> getMembers() {
-        return members;
+    public boolean isMember(UUID playerUuid) {
+        return this.members
+            .stream()
+            .filter(member -> member.getId() == playerUuid)
+            .count() == 1;
     }
 
-    public List<UUID> getMemberList() {
-        return members;
+    public ArrayList<GroupMember> getMembers() {
+        return this.members;
     }
 
-    public List<UUID> getMemberArrayList() {
-        return members;
+    public ArrayList<UUID> getMembersUUIDS() {
+        return (ArrayList<UUID>) this.members.
+            stream()
+            .map(member -> member.getId())
+            .collect(Collectors.toList());
+    }
+
+    public ArrayList<UUID> getMembersUUIDS(GroupRole role) {
+        return (ArrayList<UUID>) this.members.stream()
+            .filter(member -> member.hasRole(role))
+            .map(uuid -> uuid.getId())
+            .collect(Collectors.toList());
```

Comment:
I'd say change this to getMembers(GroupRole role) and remove any UUID getter functionality and let the end classes handle that